frontend/src/lib/components/CompareFilter/compareFilterLogic.js

```diff
@@ -26,15 +34,20 @@ export const compareFilterLogic = kea({
                 router.actions.push(pathname, searchParams)
             }
         },
-        [actions.toggleCompare]: () => {
+        toggleCompare: () => {
             actions.setCompare(!values.compare)
         },
     }),
     urlToAction: ({ actions }) => ({
-        '/insights': (_, { compare }) => {
-            if (compare) {
+        '/insights': (_, { compare, shown_as, date_from }) => {
+            if (compare != null) {
                 actions.setCompare(compare)
             }
+            if (shown_as === ShownAsValue.LIFECYCLE || date_from === 'all') {
```

Comment:
Wouldn't this make more sense as a selector rather than a separate action?

Reply:
Hmm can you pull searchparams in selector arguments? I've never done it before but if possible yeah we could move it there.

Reply:
Ah. Didn't realize it was search params - do we store them somewhere else (e.g. a related logic) - if so, connecting + selector might be cleaner. 

This is nitpicking though, all else lgtm.

Reply:
connected to the router. I tried giving it a shot actually but it wasn't really making sense to me